kernel/kernel-impl/src/main/java/org/sakaiproject/component/impl/BasicConfigurationService.java

```diff
@@ -1093,9 +1105,16 @@ protected ConfigItemImpl findConfigItem(String name, Object defaultValue) {
             }
         } else {
             if (defaultValue instanceof Number) {
-                int num = ((Number) defaultValue).intValue();
-                int intValue = this.getInt(name, num);
-                returnValue = (T) Integer.valueOf(intValue);
+                if (defaultValue instanceof Long) {
+                    long num = ((Number) defaultValue).longValue();
+                    long longValue = this.getLong(name, num);
```

Comment:
defaultvalue is a Long already so you can just case directly to Long: this.getLong(name, (Long) defaultValue);

Reply:
Thanks!